app/models/plate/creator.rb

```diff
@@ -62,20 +87,64 @@ def create_plates_from_tube_racks!(tube_racks, barcode_printer, scanned_user, _c
         add_created_plates(factory.tube_rack, [factory.plate])
       end
     end
+
+    @created_asset_group = create_asset_group(created_plates) if should_create_asset_group == 'Yes'
+
     print_job = LabelPrinter::PrintJob.new(barcode_printer.name,
                                            LabelPrinter::Label::PlateCreator,
                                            plates: created_plates.pluck(:destinations).flatten.compact,
                                            plate_purpose: plate_purpose, user_login: scanned_user.login)
 
-    unless print_job.execute
-      raise PlateCreationError, 'Barcode labels failed to print.'
-    end
-
+    warnings_list << 'Barcode labels failed to print.' unless print_job.execute
     true
   end
 
   private
 
+  def create_asset_group(created_plates)
+    group = nil
+    all_wells = created_plates.map { |hash| hash[:destinations].map(&:wells) }.flatten
+
+    study = find_relevant_study(created_plates)
+    unless study
+      warnings_list << 'Failed to create Asset Group: could not find an appropriate Study to group the plates under.'
+      return group
+    end
+
+    ActiveRecord::Base.transaction do # TO DO: handle exceptions from this?
+      group = AssetGroup.create!(study: study, name: asset_group_name)
+      group.assets.concat(all_wells)
+    end
+
+    group
+  end
+
+  def find_relevant_study(created_plates)
+    # find a relevant study to put the Asset group under
+    # otherwise would have to get user to select one
+
+    # try the link on aliquots
+    all_destination_plates = created_plates.map { |hash| hash[:destinations] }.flatten
+    study = all_destination_plates.map(&:studies).flatten.first
+    return study if study
+
+    # try the study_samples table
+    all_destination_plates.each do |plate|
+      plate.contained_samples.each do |sample|
+        return sample.studies.first if sample.studies.first
+      end
+    end
+
+    nil
+  end
+
+  def asset_group_name
+    prefix = 'plate-creator'
+    time_now_formatted = "#{Time.zone.now.year}-#{Time.zone.now.month}-#{Time.zone.now.day}-#{Time.zone.now.hour}#{Time.zone.now.min}#{Time.zone.now.sec}"
+    suffix = rand(999)
+    "#{prefix}-#{time_now_formatted}-#{suffix}"
```

Comment:
What about?
"#{prefix}-#{SecureRandom.uuid}"

Reply:
I dunno - I thought having the timestamp there might be useful if they end up having to browse the list of asset groups looking for the one they created today?